segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/metadata/StorageMetadata.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.metadata;
+
+import io.pravega.common.io.serialization.VersionedSerializer;
+import io.pravega.segmentstore.storage.chunklayer.ChunkManager;
+import io.pravega.segmentstore.storage.mocks.MockStorageMetadata;
+import lombok.Data;
+
+import java.io.Serializable;
+
+/**
+ * Storage Metadata.
+ * All storage related metadata is stored in {@link ChunkMetadataStore} using set of key-value pairs.
+ * The String value returned by {@link StorageMetadata#getKey()} is used as key.
+ * Notable derived classes are {@link SegmentMetadata} and {@link ChunkMetadata} which form the core of metadata related
+ * to {@link ChunkManager} functionality.
+ */
+@Data
+public abstract class StorageMetadata implements Serializable {
+
+    /**
+     * Retrieves the key associated with the metadata.
+     *
+     * @return key.
+     */
+    public abstract String getKey();
+
+    /**
+     * Creates a deep copy of this instance.
+     *
+     * @return
+     */
+    public abstract StorageMetadata deepCopy();
+
+    /**
+     * Helper method that converts empty string to null value.
+     *
+     * @param toConvert String to convert.
+     * @return If toConvert is null then it returns empty string. Otherwise returns original string.
+     */
+    public static String toNullableString(String toConvert) {
```

Comment:
Why don't you put these methods into "pravega common" module ?

Reply:
Not sure. Doesn't look very useful outside limited use in this class/module.